src/main/sensors/gyro_init.c

```diff
@@ -603,68 +592,69 @@ bool gyroInit(void)
         eepromWriteRequired = true;
     }
 
-#if defined(USE_MULTI_GYRO)
-    if ((gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH && !((gyroDetectionFlags & GYRO_ALL_MASK) == GYRO_ALL_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_1 && !(gyroDetectionFlags & GYRO_1_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_2 && !(gyroDetectionFlags & GYRO_2_MASK))) {
-        if (gyroDetectionFlags & GYRO_1_MASK) {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_1;
-        } else {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_2;
+    // check if all enabled sensors are detected
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        if ((gyroDetectionFlags & GYRO_MASK(i)) && (gyroConfig()->gyro_enabled_bitmask & GYRO_MASK(i))) {
+            gyro.gyroEnabledBitmask |= GYRO_MASK(i);
         }
+    }
 
-        gyroConfigMutable()->gyro_to_use = gyro.gyroToUse;
+    if (gyroConfigMutable()->gyro_enabled_bitmask != gyro.gyroEnabledBitmask) {
+        gyroConfigMutable()->gyro_enabled_bitmask = gyro.gyroEnabledBitmask;
```

Comment:
(is this desired? Single detection error will remove gyro from gyro_enabled_bitmask permanently. Maybe trust user input)

Reply:
At this line it just checks to ensure that the sensors a user has enabled have been detected, and if they haven't been detected it disables those sensors that weren't detected.

Reply:
How often do gyro not get detected?

Reply:
@ledvinap How should this logic be handled? We don't want to just check IMU that users have picked (cli methods for viewing status would be wrong when not all IMU are picked) and we don't want to check all IMU every boot (a dead or non-existant IMU will make boot slow).